src/e2e-test/java/io/cdap/plugin/bigquery/actions/GroupByActions.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.bigquery.actions;
+
+import io.cdap.e2e.pages.locators.CdfPluginPropertiesLocators;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.JsonUtils;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.plugin.bigquery.locators.GroupByLocators;
+import io.cucumber.core.logging.Logger;
+import io.cucumber.core.logging.LoggerFactory;
+import org.openqa.selenium.ElementClickInterceptedException;
+
+import java.util.Map;
+
+/**
+ * GroupBy Related Actions.
+ */
+public class GroupByActions {
+  private static final Logger logger = (Logger) LoggerFactory.getLogger(GroupByActions.class);
+
+  static {
+    SeleniumHelper.getPropertiesLocators(GroupByLocators.class);
+  }
+
+  public static void enterAggregates(String jsonAggreegatesFields) {
+    Map<String, String> fieldsMapping =
```

Comment:
Its a good practice to add a Java doc with Inline comments whenever you write a customized logic, please add the same.

Reply:
added